prefect_gcp/worker.py

```diff
@@ -0,0 +1,659 @@
+        with the `prefect -m prefect.engine` if a command is not present.
+        """
+        try:
+            command = self.job_body["spec"]["template"]["spec"]["template"]["spec"][
+                "containers"
+            ][0].get("command")
+            if command is None:
+                self.job_body["spec"]["template"]["spec"]["template"]["spec"][
+                    "containers"
+                ][0]["command"] = [
+                    "python",
+                    "-m",
+                    "prefect.engine",
+                ]
+            elif isinstance(command, str):
+                self.job_body["spec"]["template"]["spec"]["template"]["spec"][
+                    "containers"
+                ][0]["command"] = shlex.split(command)
+        except KeyError:
+            raise ValueError(
+                "Unable to verify command due to invalid job body template."
+            )
+
+    def _format_args_if_present(self):
+        try:
+            args = self.job_body["spec"]["template"]["spec"]["template"]["spec"][
+                "containers"
+            ][0].get("args")
+            if args is not None and isinstance(args, str):
+                self.job_body["spec"]["template"]["spec"]["template"]["spec"][
+                    "containers"
+                ][0]["args"] = shlex.split(args)
+        except KeyError:
+            raise ValueError("Unable to verify args due to invalid job body template.")
+
+    @validator("job_body")
+    def _ensure_job_includes_all_required_components(cls, value: Dict[str, Any]):
+        """
+        Ensures that the job body includes all required components.
+        """
+        patch = JsonPatch.from_diff(value, _get_base_job_body())
+        missing_paths = sorted([op["path"] for op in patch if op["op"] == "add"])
+        if missing_paths:
+            raise ValueError(
+                "Job is missing required attributes at the following paths: "
+                f"{', '.join(missing_paths)}"
+            )
+        return value
+
+    @validator("job_body")
+    def _ensure_job_has_compatible_values(cls, value: Dict[str, Any]):
+        """Ensure that the job body has compatible values."""
+        patch = JsonPatch.from_diff(value, _get_base_job_body())
+        incompatible = sorted(
+            [
+                f"{op['path']} must have value {op['value']!r}"
+                for op in patch
+                if op["op"] == "replace"
+            ]
+        )
+        if incompatible:
+            raise ValueError(
+                "Job has incompatible values for the following attributes: "
+                f"{', '.join(incompatible)}"
+            )
+        return value
+
+
+class CloudRunWorkerVariables(BaseVariables):
+    """
+    Default variables for the Cloud Run worker.
+
+    The schema for this class is used to populate the `variables` section of the default
+    base job template.
+    """
+
+    region: str = Field(
```

Comment:
@desertaxle is it incorrect for `region` to be a required value since there is no way to set it via `prefect work-pool create my-second-pool --type cloud-run`?

I left this as required because the infrablock had it required. But I don't see any other required fields on any other Worker setups

Reply:
I'd recommend adding a default for region to remove that friction. I think it should be fine to default to `us-central1` or a similar region.